tensorpack/tfutils/argscope.py

```diff
@@ -78,12 +81,46 @@ def wrapped_func(*args, **kwargs):
     return wrapped_func
 
 
-def enable_argscope_for_module(module):
+def input_output_shape_decorator(func):
+    """Decorator for function to support input-output-shape information
+
+    Example:
+        [0615 09:36:52 @argscope.py:90]                conv0:    [None, 28, 28, 1] ->   [None, 28, 28, 32]
+        [0615 09:36:52 @argscope.py:90]             <unkown>:   [None, 28, 28, 32] ->   [None, 14, 14, 32]
+        [0615 09:36:52 @argscope.py:90]                conv1:   [None, 14, 14, 32] ->   [None, 14, 14, 32]
+        [0615 09:36:52 @argscope.py:90]                conv2:   [None, 14, 14, 32] ->   [None, 14, 14, 32]
+        [0615 09:36:52 @argscope.py:90]             <unkown>:   [None, 14, 14, 32] ->     [None, 7, 7, 32]
+        [0615 09:36:52 @argscope.py:90]                conv3:     [None, 7, 7, 32] ->     [None, 7, 7, 32]
+        [0615 09:36:52 @argscope.py:90]             <unkown>:     [None, 7, 7, 32] ->         [None, 1568]
+        [0615 09:36:52 @argscope.py:90]                  fc0:         [None, 1568] ->          [None, 512]
+        [0615 09:36:52 @argscope.py:90]             <unkown>:          [None, 512] ->          [None, 512]
+        [0615 09:36:52 @argscope.py:90]                  fc1:          [None, 512] ->           [None, 10]
     """
-    Overwrite all functions of a given module to support argscope.
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        in_tensor = args[0]
+        out_tensor = func(*args, **kwargs)
+        name = '<unkown>' if 'name' not in kwargs else kwargs['name']
+        logger.info('%20s: %20s -> %20s' % (name, in_tensor.shape.as_list(), out_tensor.shape.as_list()))
+        return out_tensor
+    return wrapped_func
+
+
+def decorate_module(module, func):
+    """
+    Apply a decorator to all functions provided by `module`.
     Note that this function monkey-patches the module and therefore could have unexpected consequences.
     It has been only tested to work well with `tf.layers` module.
     """
     for name, obj in getmembers(module):
         if isfunction(obj):
-            setattr(module, name, argscope_mapper(obj))
+            setattr(module, name, func(obj))
+
+
+def enable_argscope_for_module(module):
```

Comment:
What about adding an option like `log_shape=True` here? Easier to use than a new API.